pallets/stateful-storage/src/stateful_child_tree.rs

```diff
@@ -42,37 +243,50 @@ impl StatefulChildTree {
 	///
 	/// Allows getting all the keys having the same prefix
 	/// Warning: This should not be used from any on-chain transaction!
-	pub fn prefix_iterator<V: Decode + Sized>(
+	pub fn prefix_iterator<
+		V: Decode + Sized,
+		K: IsStatefulKey<H> + Sized,
```

Comment:
Do you expect that different hashers are going to be used on the same type? I guess why not set this as a generic one level up?

Reply:
I'm not understanding what you're getting at... the hasher is a type parameter to `StatefulChildTree`, but it needs to make sure that the MultipartKey uses the same hasher. What am I missing?